weio/csv_file.py

```diff
@@ -59,19 +62,57 @@ def __init__(self, filename=None, sep=None, colNames=None, commentChar=None, com
         if (len(self.colNames)>0) and (self.colNamesLine is not None):
             raise Exception('Provide either `colNames` or `colNamesLine` for CSV file types')
         if filename:
-            self.read(filename, doRead=doRead, **kwargs)
+            if streaming:
+                # Don't read immediately in streaming mode - wait for context manager
+                self.filename = filename
+            else:
+                self.read(filename, doRead=doRead, **kwargs)
         else:
             self.filename = None
 
-    def read(self, filename=None, doRead=True, **kwargs):
+    def __enter__(self):

```

Comment:
I think we can rely on the parent class for the __enter__, ___exit__ and  _enforece_context_if_needed. I believe these can be removed.